pandas/core/arrays/datetimes.py

```diff
@@ -222,6 +222,10 @@ def __new__(cls, values, freq=None, tz=None, dtype=None):
         # if dtype has an embedded tz, capture it
         tz = dtl.validate_tz_from_dtype(dtype, tz)
 
+        if isinstance(values, ABCSeries):
```

Comment:
Should this be done immediately? A `Series[datetime64[ns, tz]]` will not have a `.tz` attribute.

On this branch:

```python
In [16]: ser = pd.Series(pd.date_range('2017', periods=4, tz='UTC'))

In [17]: pd.core.arrays.DatetimeArrayMixin(ser).tz

In [18]: pd.core.arrays.DatetimeArrayMixin(ser._values).tz
Out[18]: <UTC>
```

I suspect we're never hitting this case (passing a series but not a dtype or `tz`), but fixing it now may save some future debugging.

Reply:
Good catch